app/controllers/reports_controller.rb

```diff
@@ -48,10 +43,42 @@ def create
 
     rep.save
 
-    render json: {id: rep.slug}
+    render json: response_hash(rep)
+  rescue => err
+    logger.fatal("Error: #{err.message} || #{data}")
+    head 400
+    nil
   end
 
   def show
     @report = Report.find_by slug: params[:id]
   end
+
+  private
+
+  def create_report(project, entries)
+    if project
+      project.reports.create report: JSON.generate(entries)
+    else
+      Report.create report: JSON.generate(entries)
+    end
+  end
+
+  def find_project(input)
+    return if input["project"].blank?
+
+    if input["project"]["id"]
+      Project.find(input["project"]["id"])
+    elsif input["project"]["permalink"]
+      Project.find_or_create_by(permalink: input["project"]["permalink"])
```

Comment:
@etagwerker Do we want the `permalink` field to be unique?

Reply:
@lubc Sounds good 👍